### ISSUE TITLE
Queue Settings Issues.

### ISSUE THREAD

**Issue by eraz** [vidalborromeo/eecshelp-beta#8]
v2.30

A change to queue requirements is applied immediately, but not to form submission fields. e.g. If the queue required users to provide a location, a user must still fill out the field even after the requirement is disabled until they reload the page. Likewise, if the queue did not require users to provide a location, a user can still submit a request without a location after the requirement is enabled, but will get a "Bad Request" error.

Also, the queue settings menu *appears* to remain unchanged after being updated until the page is refreshed. e.g. If the requirement to provide a location is disabled, the box will still be checked when reopening the settings menu, but users will not need to provided a location (unless they have not reloaded their pages).

Updating the queue settings in any manner (pressing the save button) immediately closes any new request forms in other sessions (forcing the user to start over). If a user was editing a request, the form will disappear and **leave the page in an non-interactive state**, forcing the user to refresh the page.

TL;DR - Updates to queue settings must be accompanied by a page refresh in any session (student or instructor) for all changes to have their full and appropriate effects.

**Comment by vidalborromeo** [comment]
Grrr. This one has me confuzzled. I've triple-checked each step in my code and the values being operated on. There must be something sinister going on in the bowels of the browser's DOM. Will keep at it.

**Comment by vidalborromeo** [comment]
UPDATE:
After much banging of head against wall, I've discerned that this is actually _three_ distinct problems. I'll fix all three in the same commit.

For my own future reference:
- The setInSlot() method automatically triggers onReset() for all parent presenters in the tree from the current presenter. To avoid this, use the 3-arg version.

**Comment by vidalborromeo** [comment]
Fixed in v2.41

**Comment by vidalborromeo** [comment]
@eraz, can you cofirm?

**Comment by eraz** [comment]
New Request submission form is still force-closed in other sessions upon queue settings change.
Request Edit form no longer force-closes and the site remains responsive. :+1: 
Required form fields are updated in real-time and correctly in the Request Edit form. :+1: :+1: :+1:
